File: Prime/Perl/Prime.pl
sub Prime {
	if ($_[0] < 2) {
		return -1;
	}
	if ($_[0] == 2) {
		return 1;
	}
	if ($_[0]%2 == 0) {
		return 0;
	}
	for ($i = 3; $i <= sqrt($_[0]); $i+=2) {
		if ($_[0]%$i == 0) {
			return 0;
		}
	}
	1;
}

sub PrintPrime {
	if ($_[0] < 0) {
		return "Error";
	}
	if ($_[0] == 0) {
		return "Composite number";
	}
	"Prime number";
}

while(1) {
	print "number : ";
	my $n = <STDIN>;
	if ($n < 0) {
	  $n = 0;
	}
	print &PrintPrime(&Prime($n))."\n\n";
	print("Continue? (Yes : 1 / No : 0) : ");
	my $i = <STDIN>;
	print "\n";
	if($i == 0) {
		last;
	}
}

File: Prime/Ruby/Prime.rb
def Prime(x)
	return -1 if x < 2
	return 1 if x == 2
	return 0 if x%2 == 0
	3.step(Integer.sqrt(x),2) {|i|
		return 0 if x%i == 0
	}
	return 1
end

def PrintPrime(x)
	return "Error" if x < 0
	return "Composite number" if x == 0
	return "Prime number"
end

loop {
	print("number : ")
	n = gets.chomp.to_i
	if n < 0 then
		n = 0
	end
	puts(PrintPrime(Prime(n)))
	puts("")
	print("Continue? (Yes : 1 / No : 0) : ")
	i = gets.chomp.to_i
	puts("")
	break if i == 0
}